src/cilium_manifests.py

```diff
@@ -197,3 +212,36 @@ def config_hash(self) -> str:
         hash = hashlib.sha256()
         hash.update(json_str.encode())
         return hash.hexdigest()
+
+    @contextlib.contextmanager
+    def restart_if_exists(self):
+        ciliumDS = None
+        try:
+            ciliumDS = self.client.get(DaemonSet, name="cilium", namespace="kube-system")
+        except httpx.ConnectTimeout:
+            pass
+
+        yield
+
+        if not ciliumDS:
+            return
+
+        with IPRoute() as ip:
+            try:
+                idx = ip.link_lookup(ifname="cilium_vxlan")
+                if len(idx) > 0:
+                    ip.link("del", index=idx[0])
+            except Exception:
+                log.exception("Error in removing the cilium interface")
+
+        now = datetime.datetime.now()
+        now = str(now.isoformat("T") + "Z")
```

Comment:
nit:

python `.now()` is timezone sensative based on the machines timezone, but doesn't encoded the TZ in the object.   Similarly is `.utcnow()` but it also doesn't encode the timezone in the datetime object

The right way is like this which should generate a nice timezone string but rather than use the acceptable `+00:00` encoding for UTC, you CAN replace it with "Z" if you like

```suggestion
        now = datetime.now(timezone.utc).isoformat().replace("+00:00", "Z")
```

Reply:
to use this suggestion,

```
from datetime import datetime, timezone
```

Reply:
Nice, thank you
Done